### PR TITLE
Add DB skeleton + OAuth login

### PR DESCRIPTION
**Pull request by sxmichaels**
This PR adds the following
* A skeletal DB structure with user, student, room, suite, and dorm objects
* Basic pages for adding/removing/editing the above
* Authorization policy via [https://github.com/elabs/pundit/](pundit), right now just for User, to serve as an example
 * Google OAuth login. Users will only be allowed to login if a user with a corresponding email already exists. So, right now **you will need to manually create a user with your own email** via `rails console`; enter `User.create :first_name => "First", :last_name => "Last", :email => "your_email@g.hmc.edu", :is_ashmc_admin => true, :is_super_admin => true`. This will let you access `/users/new` and create other users.
* Absolutely no nice error checking! If you fail to authenticate or something like that, you'll just get a Ruby error. Prettification is left to the UI and Admin (?) teams.
* Some basic tests for user records

# Routes
The full list of routes added by this PR is as follows.
```
      Prefix Verb   URI Pattern                        Controller#Action
       users GET    /users(.:format)                   users#index
             POST   /users(.:format)                   users#create
    new_user GET    /users/new(.:format)               users#new
   edit_user GET    /users/:id/edit(.:format)          users#edit
        user GET    /users/:id(.:format)               users#show
             PATCH  /users/:id(.:format)               users#update
             PUT    /users/:id(.:format)               users#update
             DELETE /users/:id(.:format)               users#destroy
       dorms GET    /dorms(.:format)                   dorms#index
             POST   /dorms(.:format)                   dorms#create
    new_dorm GET    /dorms/new(.:format)               dorms#new
   edit_dorm GET    /dorms/:id/edit(.:format)          dorms#edit
        dorm GET    /dorms/:id(.:format)               dorms#show
             PATCH  /dorms/:id(.:format)               dorms#update
             PUT    /dorms/:id(.:format)               dorms#update
             DELETE /dorms/:id(.:format)               dorms#destroy
       rooms GET    /rooms(.:format)                   rooms#index
             POST   /rooms(.:format)                   rooms#create
    new_room GET    /rooms/new(.:format)               rooms#new
   edit_room GET    /rooms/:id/edit(.:format)          rooms#edit
        room GET    /rooms/:id(.:format)               rooms#show
             PATCH  /rooms/:id(.:format)               rooms#update
             PUT    /rooms/:id(.:format)               rooms#update
             DELETE /rooms/:id(.:format)               rooms#destroy
      suites GET    /suites(.:format)                  suites#index
             POST   /suites(.:format)                  suites#create
   new_suite GET    /suites/new(.:format)              suites#new
  edit_suite GET    /suites/:id/edit(.:format)         suites#edit
       suite GET    /suites/:id(.:format)              suites#show
             PATCH  /suites/:id(.:format)              suites#update
             PUT    /suites/:id(.:format)              suites#update
             DELETE /suites/:id(.:format)              suites#destroy
             GET    /auth/:provider/callback(.:format) sessions#create
auth_failure GET    /auth/failure(.:format)            redirect(301, /)
      logout GET    /logout(.:format)                  sessions#destroy
    sessions POST   /sessions(.:format)                sessions#create
     session DELETE /sessions/:id(.:format)            sessions#destroy
       login GET    /login(.:format)                   logins#show
             GET    /users(.:format)                   users#index
             POST   /users(.:format)                   users#create
             GET    /users/new(.:format)               users#new
             GET    /users/:id/edit(.:format)          users#edit
             GET    /users/:id(.:format)               users#show
             PATCH  /users/:id(.:format)               users#update
             PUT    /users/:id(.:format)               users#update
             DELETE /users/:id(.:format)               users#destroy
             GET    /rooms(.:format)                   rooms#index
             POST   /rooms(.:format)                   rooms#create
             GET    /rooms/new(.:format)               rooms#new
             GET    /rooms/:id/edit(.:format)          rooms#edit
             GET    /rooms/:id(.:format)               rooms#show
             PATCH  /rooms/:id(.:format)               rooms#update
             PUT    /rooms/:id(.:format)               rooms#update
             DELETE /rooms/:id(.:format)               rooms#destroy
             GET    /suites(.:format)                  suites#index
             POST   /suites(.:format)                  suites#create
             GET    /suites/new(.:format)              suites#new
             GET    /suites/:id/edit(.:format)         suites#edit
             GET    /suites/:id(.:format)              suites#show
             PATCH  /suites/:id(.:format)              suites#update
             PUT    /suites/:id(.:format)              suites#update
             DELETE /suites/:id(.:format)              suites#destroy
             GET    /dorms(.:format)                   dorms#index
             POST   /dorms(.:format)                   dorms#create
             GET    /dorms/new(.:format)               dorms#new
             GET    /dorms/:id/edit(.:format)          dorms#edit
             GET    /dorms/:id(.:format)               dorms#show
             PATCH  /dorms/:id(.:format)               dorms#update
             PUT    /dorms/:id(.:format)               dorms#update
             DELETE /dorms/:id(.:format)               dorms#destroy
        root GET    /                                  login#show
```

### CODE REVIEW

**Comment by g-rocket** [comment]
I haven't given this a full in-depth review, but it looks reasonable to me.  Does everything in here work to the degree that other teams can use it and build upon it?  If so, I'd say merge it (once it's been more fully reviewed); if not, I'd hold off until everything is usable or try to merge the subset that is.

File: README.md [hunk 1]
```diff
@@ -1,24 +1,8 @@
...
-
-* Services (job queues, cache servers, search engines, etc.)
-
-* Deployment instructions
-
-* ...
+This app uses Google OAuth for login. You need a secret key for the Google Apps
+API, which the app will load automatically from a file named `.env` in the root
+directory. We can't commit the keys to GitHub; ask your PM for these. I sent out
+the contents of `.env` over Slack. -- Spencer
```
Comment (g-rocket) [inline README.md]: Can we just send the keys to #general so everyone has them?

**Comment by sxmichaels** [comment]
OK, this should basically be ready to merge now — you all should be able to work off of what's here to do auth/account and room/suite/dorm related stuff.

File: README.md (hunk 1 above)
Comment (sxmichaels) [reply README.md]: Done.

**Comment by jesswinssinger** [comment]
If there are no other motions, I'm going to go ahead and merge so we can close this pull request!

**Pull request merged**